ci-tools/fpga-boss/src/main.rs

```diff
@@ -301,10 +346,14 @@ fn main_impl() -> anyhow::Result<()> {
             }
         }
         Some(("serve", sub_matches)) => {
+            // Reuse the previous token if we never connect to GitHub.
+            // This avoids creating a bunch of offline runners if the FPGA fails to establish a
+            // connection.
+            let mut cached_token: Option<Vec<u8>> = None;
```

Comment:
Once the VCK-190 is "stable" I would like to remove the caching logic and the loop label, in favor of failing and letting a supervisor process restart the serve command.